src/hotspot/cpu/x86/vm_version_x86.cpp

```diff
@@ -1900,7 +2579,46 @@ void VM_Version::initialize() {
   detect_virt_stub = CAST_TO_FN_PTR(detect_virt_stub_t,
                                      g.generate_detect_virt());
 
-  get_processor_features();
+  assert(      _features == 0,       "_features should be zero at startup");
+  assert(_glibc_features == 0, "_glibc_features should be zero at startup");
+  get_processor_features_hardware();
+
+  assert(!CPUFeatures == FLAG_IS_DEFAULT(CPUFeatures), "CPUFeatures parsing");
+  uint64_t GLIBCFeatures_x64;
+  uint64_t   CPUFeatures_x64 = CPUFeatures_parse(GLIBCFeatures_x64);
+  uint64_t       features_missing =   CPUFeatures_x64 & ~      _features;
+  uint64_t glibc_features_missing = GLIBCFeatures_x64 & ~_glibc_features;
+  if (features_missing || glibc_features_missing) {
+    static const char part1[] = "Specified -XX:CPUFeatures=";
+    tty->print_raw(part1, sizeof(part1) - 1);
+    nonlibc_tty_print_uint64_comma_uint64(CPUFeatures_x64, GLIBCFeatures_x64);
+    static const char part2[] = "; this machine's CPU features are ";
+    tty->print_raw(part2, sizeof(part2) - 1);
+    nonlibc_tty_print_uint64_comma_uint64(_features, _glibc_features);
+    fatal_missing_features(features_missing, glibc_features_missing);
+  }
+
+  uint64_t       features_saved =       _features;
+  uint64_t glibc_features_saved = _glibc_features;
+
+        _features =   CPUFeatures_x64;
+  _glibc_features = GLIBCFeatures_x64;
+
+  if (ShowCPUFeatures)
+    nonlibc_tty_print_using_features_cr();
```

Comment:
Can be a regular printing function call?

Reply:
In fact it was already using a regular printing function (`print_cr()`). It was just a wrongly named function, renamed it to: `print_using_features_cr()`